codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -124,6 +125,66 @@ update these file paths accordingly.
     await tempDir.delete(recursive: true);
   }
 
+  /// Retrieve eninge artifact from google cloud storage and kick start a
+  /// recursive visit of its contents.
+  ///
+  /// Invokes [visitDirectory] to recursively visit the contents of the remote
+  /// zip. Also downloads, notarizes and uploads the engine artifact.
+  Future<void> processRemoteZip({
+    required String artifactFilePath,
+    required Directory parentDirectory,
+  }) async {
+    final FileSystem fs = tempDir.fileSystem;
```

Comment:
Why is tempDir a class field? You can do `fileSystem.systemTempDirectory.createTempSync()`

https://api.flutter.dev/flutter/package-file_file/Directory/createTempSync.html

Reply:
Right yeah the naming is probably bad, I renamed tempDir to rootDirectory. I treated tempDir as the root directory for file processing. so that api calls such as `visitZip` `processZip` that need to create folder relative to the root directory, could access the root directory through class field. Otherwise we would need to pass root directory information (formerly tempDir) as a parameter to the api calls.

Reply:
I'm still confused. Generally, you should only need the `Directory` or `FileSystem`. Why do you need both?

Reply:
Right, yeah I am using `Directory` for folders related creation in the api calls, like rootDirectory.childDirectory(something). So api calls can share the instance of the rootDirectory, and create/delete directories relative to the root directory.

umm my understanding is that `FileSystem` is an abstract class, and to use utilities like fileSystem.path.join and others, we need to invoke based on an instance of the fileSystem. I created an instance of the `FileSystem` as a class field, so api calls can share this instance and invoke the filesystem related utility functions.